package/src/activation.ts

```diff
@@ -0,0 +1,470 @@
+            pos: sourcegraph.Position,
+            ctx: sourcegraph.ReferenceContext
+        ) =>
+            // TODO - add memoizePrevious, abortPrevious
+            observableFromAsyncGenerator(() =>
+                externalReferencesProvider.references(doc, pos, ctx)
+            ),
+    }
+}
+
+function registerImplementationsProvider(
+    ctx: sourcegraph.ExtensionContext,
+    selector: sourcegraph.DocumentSelector,
+    implementationsProvider: ImplementationsProvider
+) {
+    ctx.subscriptions.add(
+        sourcegraph.languages.registerLocationProvider(
+            implementationsProvider.implId,
+            selector,
+            {
+                provideLocations: (
+                    doc: sourcegraph.TextDocument,
+                    pos: sourcegraph.Position
+                ) =>
+                    // TODO - add memoizePrevious, abortPrevious
+                    observableFromAsyncGenerator(() =>
+                        implementationsProvider.locations(doc, pos)
+                    ),
+            }
+        )
+    )
+
+    const IMPL_ID = implementationsProvider.implId
+    const panelView = sourcegraph.app.createPanelView(IMPL_ID)
+    panelView.title = implementationsProvider.panelTitle
+    panelView.component = { locationProvider: IMPL_ID }
+    panelView.priority = 160
+    ctx.subscriptions.add(panelView)
+}
+
+//
+//
+//
+
+export const areProviderParamsEqual = (
+    [doc1, pos1]: [sourcegraph.TextDocument, sourcegraph.Position],
+    [doc2, pos2]: [sourcegraph.TextDocument, sourcegraph.Position]
+): boolean => doc1.uri === doc2.uri && pos1.isEqual(pos2)
+
+export const observableFromAsyncGenerator = <T>(
+    generator: () => AsyncGenerator<T, unknown, void>
+): Observable<T> =>
+    new Observable((observer: Observer<T>) => {
+        const iterator = generator()
+        let unsubscribed = false
+        let iteratorDone = false
+        function next(): void {
+            iterator.next().then(
+                result => {
+                    if (unsubscribed) {
+                        return
+                    }
+                    if (result.done) {
+                        iteratorDone = true
+                        observer.complete()
+                    } else {
+                        observer.next(result.value)
+                        next()
+                    }
+                },
+                err => {
+                    observer.error(err)
+                }
+            )
+        }
+        next()
+        return () => {
+            unsubscribed = true
+            if (!iteratorDone && iterator.throw) {
+                iterator.throw(createAbortError()).catch(() => {
+                    // ignore
+                })
+            }
+        }
+    })
+
+/**
+ * Similar to Rx `switchMap`, finishes the generator returned from the previous call whenever the function is called again.
+ *
+ * Workaround for https://github.com/sourcegraph/sourcegraph/issues/1190
```

Comment:
The issue linked has been closed for over a year, so this probably shouldn't be lugged around to this repo

Reply:
Looks like we can remove `abortPrevious` but need to keep `memoizePrevious` due to https://github.com/sourcegraph/sourcegraph/issues/1321?

Reply:
Removed.